src/llama_stack_client/lib/agents/agent.py

```diff
@@ -49,40 +73,27 @@ async def create_turn(
             attachments=attachments,
             stream=True,
         )
-        turn = None
         for chunk in response:
-            if chunk.event.payload.event_type != "turn_complete":
+            if not self._has_tool_call(chunk):
                 yield chunk
-            else:
```

Comment:
what happens in this branch if there _is_ a tool call?

Reply:
A little bit unclear right now:
- create_turn --> don't execute tool calls
- async_create_turn --> execute tool calls (async execution)

Was thinking we should have `CustomTool` & `AsyncCustomTool` classes. But I don't want to add too many extra unused/untested code in the SDK at the moment.

Reply:
Talked offline. Let's make CustomTool calls synchronous for now and hoist up the responsibility of thread pool management entirely to the app layer if create_turn() ends up being too blocking in nature.

We should eventually figure out how AsyncCustomTool should work with AsyncLlamaStackClient

Reply:
synced offline, making all tool execution synchrounous